crud_handler/crud_handler.go

```diff
@@ -160,18 +163,30 @@ func (h *Handler) GetAllRecords(w http.ResponseWriter, r *http.Request) {
 func (h *Handler) GetRecord(w http.ResponseWriter, r *http.Request) {
 	id := chi.URLParam(r, "id")
 	// fmt.Printf("id = %s \n", id)
+
+	res := h.cache.Get(id)
```

Comment:
I guess there is a good case when we need to define the service level and encapsulate business logic in it.

How it should look like: 
1. handler parses a request into the model that the service accepts
2. service does all logic and returns some response
3. handler converts the service's response into output representation format (json, HTML, grpc, etc.)

Create a separate package called `service` and move all business logic there. As well as all dependencies as cache or db, etc.